contracts/Dependencies/Mynt/MyntLib.sol

```diff
@@ -29,10 +29,14 @@ library MyntLib {
         address _toToken,
         IMassetManager.PermitParams calldata _permitParams
     ) internal returns (uint256) {
+        IDLLR dllrTransferWithPermit = IDLLR(_myntMassetManager.mAssetTokenTransferWithPermit());
         IDLLR dllr = IDLLR(_myntMassetManager.getToken());
+
         uint256 thisBalanceBefore = dllr.balanceOf(address(this));
         address thisAddress = address(this);
-        dllr.transferWithPermit(
+
+        IDLLR tokenTransferWithPermit = address(dllrTransferWithPermit) != address(0) ? dllrTransferWithPermit : dllr;
```

Comment:
nice solution, but considering that FE needs to know which contract to ask permission for in advance, and querying the address on each call is an overkill, pls consider the following redesign:
1. save original function with name `redeemZusdDirectlyFromDllrWithPermit`, but reverting with the message - "use redeemZusdFromDllrWithPermit function" and comment that this revert can be removed, the function redeemZusdFromDllrWithPermit removed and `redeemZusdDirectlyFromDllrWithPermit` renamed to `redeemZusdFromDllrWithPermit` only if the aggregator token (DLLR) has no griefing attack vulnerablility and a link to the commit with this fix for clarity on how to check it
2. replace dllr with dllrTransferWithPermit and add a comment that Permit should be issued for the contract dllrTransferWithPermit